typescript/sdk/src/providers/transactions/submitter/IcaTxSubmitter.ts

```diff
@@ -0,0 +1,139 @@
+import { IRegistry } from '@hyperlane-xyz/registry';
+import { Address, ProtocolType, assert } from '@hyperlane-xyz/utils';
+
+import {
+  InterchainAccount,
+  buildInterchainAccountApp,
+} from '../../../middleware/account/InterchainAccount.js';
+import { MultiProvider } from '../../MultiProvider.js';
+import {
+  AnnotatedEV5Transaction,
+  ProtocolTypedReceipt,
+} from '../../ProviderType.js';
+import { CallData } from '../types.js';
+
+import { TxSubmitterInterface } from './TxSubmitterInterface.js';
+import { TxSubmitterType } from './TxSubmitterTypes.js';
+import { getSubmitter } from './submitterBuilderGetter.js';
+import { EvmIcaTxSubmitterProps } from './types.js';
+
+type EvmIcaTxSubmitterConfig = EvmIcaTxSubmitterProps & {
+  originInterchainAccountRouter: Address;
+};
+
+type EvmIcaTxSubmitterConstructorConfig = Omit<
+  EvmIcaTxSubmitterConfig,
+  'internalSubmitter' | 'type'
+>;
+
+export class EvmIcaTxSubmitter
+  implements TxSubmitterInterface<ProtocolType.Ethereum>
+{
+  readonly txSubmitterType: TxSubmitterType =
+    TxSubmitterType.INTERCHAIN_ACCOUNT;
+
+  protected constructor(
+    protected readonly config: EvmIcaTxSubmitterConstructorConfig,
+    protected readonly submitter: TxSubmitterInterface<ProtocolType.Ethereum>,
+    protected readonly multiProvider: MultiProvider,
+    protected readonly interchainAccountApp: InterchainAccount,
+  ) {}
+
+  static async fromConfig(
+    config: EvmIcaTxSubmitterConfig,
+    multiProvider: MultiProvider,
+    registry: Readonly<IRegistry>,
+  ): Promise<EvmIcaTxSubmitter> {
+    const internalSubmitter = await getSubmitter<ProtocolType.Ethereum>(
+      multiProvider,
+      config.internalSubmitter,
+      registry,
+    );
+
+    const interchainAccountApp: InterchainAccount =
+      await buildInterchainAccountApp(
+        multiProvider,
+        config.chain,
+        {
+          owner: config.owner,
+          origin: config.chain,
+          localRouter: config.originInterchainAccountRouter,
+          ismOverride: config.interchainSecurityModule,
+        },
+        registry,
+      );
+
+    return new EvmIcaTxSubmitter(
+      {
+        owner: config.owner,
+        chain: config.chain,
+        destinationChain: config.destinationChain,
+        originInterchainAccountRouter: config.originInterchainAccountRouter,
+      },
+      internalSubmitter,
+      multiProvider,
+      interchainAccountApp,
+    );
+  }
+
+  async submit(
+    ...txs: AnnotatedEV5Transaction[]
+  ): Promise<
+    | void
+    | ProtocolTypedReceipt<ProtocolType.Ethereum>['receipt']
+    | ProtocolTypedReceipt<ProtocolType.Ethereum>['receipt'][]
+  > {
+    if (txs.length === 0) {
+      return [];
+    }
+
+    const transactionChains = new Set(txs.map((tx) => tx.chainId));
+    if (transactionChains.size !== 1) {
+      throw new Error(
+        'ICA transactions should have all the same destination chain',
```

Comment:
isnt this the origin chain according to the example

Reply:
Noup, because they transactions without the ica submitter would have been sent from the chain initially specified in each one. Imagine this:
when you need to perform an update to a warp route token on sepolia all the update transaction will have as destination chain sepolia, but if you send those transactions with an ica from those transactions will be sent from another chain (basesepolia for examaple), but the intended destination chain to receive those transactions will always be sepolia

Reply:
If you see below, after asserting that all the transactions are directed to the same chain we compare the destination chain name of the txs with the destination chain name configured on this ica submitter